### PR TITLE
RD-3007 execution graph: hide new builtin workflow tasks

### PR DESCRIPTION
**Pull request by tehasdf**
Ever since cloudify-cosmo/cloudify-common#808 the builtin tasks
are no longer LocalWorkflowTasks, but have their own type. To keep
them hidden, specialcase those types as well.

## Screenshots / Videos
Example before/after with a node-instance with one operation
before
after

## Checklist

- [ ] My code follows the UI Code Style.
- [ ] I verified that all tests and checks have passed.
- [x] I verified if that change requires a documentation update.
- [ ] I added proper labels to this PR.

## Tests
Tested manually by running an execution and examining the graph.
This widget has no automated tests and I'm sure as hell not adding
them while making a change this trivial.

## Documentation
N/A

### CODE REVIEW

File: widgets/executions/src/backend.js [hunk 1]
```diff
@@ -222,6 +222,23 @@ module.exports = r => {
...
+                'NOPLocalWorkflowTask',
+            ];
+            return (
+                workflowTask.labels[0].retry > 0 ||
+                _.includes(typesToHide, workflowTask.labels[0].type)
+            );
+        };
         const safeDeleteIrrelevantGraphVertices = allSubgraphs => {
             // Remove LocalWorkflow, NOPWorkflowTasks and retrying-tasks from the graph
             // while keeping it connected
```
Comment (qooban) [inline widgets/executions/src/backend.js:244]: Maybe worth to update that comment?

File: widgets/executions/src/backend.js [hunk 2]
```diff
@@ -232,11 +249,7 @@ module.exports = r => {
...
                         // For each subgraph, go through all the tasks
-                        if (
-                            workflowTask.labels[0].type === localWorkflowTask ||
-                            workflowTask.labels[0].type === nopLocalWorkflowTask ||
-                            workflowTask.labels[0].retry > 0
-                        ) {
+                        if (shouldHideTask(workflowTask)) {
                             // Remove all LocalWorkflowTasks and NOPWorkflowTasks from the subgraph
                             // Connect its 'target' edges to it's parents' 'target' edges
                             // Remove the node when done
```
Comment (qooban) [inline widgets/executions/src/backend.js:255]: And that comment should be updated as well, I think.

File: widgets/executions/src/backend.js [hunk 3]
```diff
@@ -232,11 +249,7 @@ module.exports = r => {
                     // eslint-disable-next-line consistent-return
                     subGraph.children = _.map(subGraph.children, workflowTask => {
                         // For each subgraph, go through all the tasks
-                        if (
-                            workflowTask.labels[0].type === localWorkflowTask ||
-                            workflowTask.labels[0].type === nopLocalWorkflowTask ||
```
Comment (qooban) [inline widgets/executions/src/backend.js:237]: Seems like after that change we don't need:
```
        const localWorkflowTask = 'LocalWorkflowTask';
        const nopLocalWorkflowTask = 'NOPLocalWorkflowTask';
```
from L79 - am I correct?

**Review by qooban — COMMENTED** [review]
Added some comments.

Regarding tests, there is `executions_spec.ts`, so I think you can trigger Stage system tests: https://jenkins.cloudify.co/view/UI%20Health%20View/job/Stage-UI-System-Test/build?delay=0sec on `exec-graph-hide-new-task-types-1` as `STAGE_BRANCH` with `test/cypress/integration/widgets/executions_spec.ts` as `TEST_SPEC` to verify.

Documentation is not needed, I think, as we don't specify hidden tasks anywhere.

**Comment by tehasdf** [comment]
@qooban I know there's a spec but it doesn't test the graph and as I noted, I'm not gonna add tests for a change that is essentially changing a string task.type to another

**Review by qooban — APPROVED** [review]
Looks good 👍🏻 

Just one small linter issue to fix:
```
[2021-08-19T12:20:48.669Z] /home/jenkins/agent/workspace/exec-graph-hide-new-task-types-1/test/widgets/executions/src/backend.js
[2021-08-19T12:20:48.669Z]   232:39  error  Delete `,`  prettier/prettier
```

**Review by qooban — APPROVED** [review]

**Pull request merged**